src/vs/workbench/services/editor/browser/editorService.ts

```diff
@@ -127,7 +127,7 @@ export class EditorService extends Disposable implements EditorServiceImpl {
 			return; // empty groups do not change structure of editors
 		}
 
-		this._onDidEditorsChange.fire();
+		this._onDidEditorsChange.fire({ groupId: group.id, kind: GroupChangeKind.GROUP_INDEX });
```

Comment:
I think this entire listener is actually not needed anymore because you are actually releasing all `GroupChangeKind` events via the code below already.